Microsoft.Toolkit.Uwp.UI.Animations/Behaviors/ViewportBehavior.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media;
+
+namespace Microsoft.Toolkit.Uwp.UI.Animations.Behaviors
+{
+    /// <summary>
+    /// A class for listening element enter or exit the viewport
+    /// </summary>
+    public class ViewportBehavior : BehaviorBase<FrameworkElement>
```

Comment:
Should we change it to be based on `ScrollViewer` instead of `FrameworkElement`?

Reply:
No. ```FrameworkElement``` is right.
Take an example:
```XAML
<Canvas x:Name="RootCanvas">
    <Image Canvas.Left="123"
           Canvas.Top="666"
           Width="50"
           Height="50">
        <i:Interactions.Behavior>
            <toolkit:ViewportBehavior ExitingViewport="Image_ExitingViewport" />
        </i:Interaction.Behavior>
    </Image>
</Canvas>
```
At begin, the window is 800x600. After resize, the window change to 400x300, then we will got the exit event.

Reply:
At least it should be using `SizeChanged` not `LayoutUpdated`. And most of the time this is tied with a `ScrollViewer`, where `ViewChanged` is a much better trigger event IMO.

Reply:
Done, use ```ViewChanged``` instead ```LayoutUpdated```, but it only works the element under a ```ScrollViewer```. Can you review the codes again?

Reply:
Do you think if it's a good idea to have a bool flag to unsubscribe the `ViewChanged` event when the element has completely entered the viewport? Because most of the time once we show the image, we don't care if it's entering or leaving the viewport anymore.

Reply:
Done